### PR TITLE
Moving DataGridView ToolTip test to the MAUI test suite

### PR DESCRIPTION
**Pull request by ArtemTatarinov**
## Proposed changes

- To minimize unit tests that have dependencies on global singletons like mouse, move the test for `DataGridView` `Tooltip` from unit tests to the MAUI test suite, so race conditions wouldn't occur.
- This test is rather specific: there should be a separate `Form` to reproduce the issue, not the default MAUI test `Form` inherited from `ReflectBase` class.

## Customer Impact

- No

## Regression? 

- No

## Risk

- Minimal

## Test environment(s)
Microsoft Windows [Version 10.0.19043.1165]
.NET SDK 6.0.100-rc.1.21416.40

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5639)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using ReflectTools;
+using WFCTestLib.Log;
+
+namespace System.Windows.Forms.IntegrationTests.MauiTests
+{
+    public class MauiDataGridViewTests : ReflectBase
+    {
+        public MauiDataGridViewTests(string[] args) : base(args)
+        {
+            this.BringToForeground();
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs:16]: nit: remove `this`

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        }
+
+        public static void Main(string[] args)
+        {
+            Thread.CurrentThread.SetCulture("en-US");
+            Application.Run(new MauiDataGridViewTests(args));
+        }
+
+        // This test is rather specific: there should be a separate Form to reproduce the issue.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs:25]: Could you please explain in this comment why a separate forms is needed?

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            // Create DataGridView with DataSource set to dataTable.
+            using DataGridView dataGridView = new();
+            dataGridView.ShowCellToolTips = true;
+            dataGridView.DataSource = dataTable;
+
+            // Create form and add dataGridView.
+            using Form dialog = new();
+            dialog.Controls.Add(dataGridView);
+
+            // These two steps can also be done manually after the dialog is shown.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs:42]: Is this comment needed?

File: src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiDataGridViewTests.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    ///  the scenarios once and store the results, then feed the scenario names in as member data.
+    ///
+    ///  However, MemberData is resolved before any constructors (even static) are called.
+    ///  This means the scenario names will not be available yet.
+    ///
+    ///  The solution to this is to inherit from MemberDataAttribute and execute custom code
+    ///  (running the maui test) before returning the expected data. See MauiMemberDataAttribute.cs for more info.
+    ///
+    ///  Also [Collection("Maui")] is used put all maui tests in the same collection, which makes them run sequentially
+    ///  instead of in parallel. This is to mitigate race conditions of multiple forms open at once.
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiDataGridViewTests.cs:23]: `.. to mitigate race conditions when accessing singleton resources such as clipboard or mouse`?

**Review by Tanya-Solyanik — COMMENTED** [review]
Looks good, I added a couple of nits.

**Comment by dreddy-work** [comment]
We need this updated on RC2 PR as well if not done already.

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs (hunk 1 above)
Comment (ArtemTatarinov) [reply src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs:16]: MAUI test won't build without `this`:
image

Documentation also tells to use `this`:
image

**Review by dreddy-work — APPROVED** [review]

File: src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs (hunk 1 above)
Comment (Tanya-Solyanik) [reply src/System.Windows.Forms/tests/IntegrationTests/MauiTests/MauiDataGridViewTests/MauiDataGridViewTests.cs:16]: I missed that this was a constructor.

**Review by Tanya-Solyanik — APPROVED** [review]
The change looks good, but please follow Devendars guidance regarding merging and cherry picking

**Pull request merged**